Error detected in test_cluster_performance

In this issue we will investigate and fix a bug detected during the launch of the tests for the PR https://github.com/wazuh/wazuh-qa/pull/2758. The error is the following:

```
pytest -s tests/performance/test_cluster/test_cluster_performance/test_cluster_performance.py --artifacts_path=/home/adriiiprodri/artifacts --n_workers=10 --n_agents=50000
============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.9.5, pytest-6.2.5, py-1.10.0, pluggy-0.13.1
rootdir: /home/adriiiprodri/git/wazuh-qa
plugins: asyncio-0.15.1, trio-0.7.0, cov-2.12.0, metadata-1.11.0, tavern-1.19.0, testinfra-6.4.0, html-3.1.1, aiohttp-0.3.0
collected 1 item

tests/performance/test_cluster/test_cluster_performance/test_cluster_performance.py F

==================================================================================================== FAILURES ====================================================================================================
____________________________________________________________________________________________ test_cluster_performance ____________________________________________________________________________________________

artifacts_path = '/home/adriiiprodri/artifacts', n_workers = 10, n_agents = 50000

    def test_cluster_performance(artifacts_path, n_workers, n_agents):
        """Check that a cluster environment did not exceed certain thresholds.

        This test obtains various statistics (mean, max, regression coefficient) from CSVs with
        data generated in a cluster environment (resources used and duration of tasks). These
        statistics are compared with thresholds established in the data folder.

        Args:
            artifacts_path (str): Path where CSVs with cluster information can be found.
            n_workers (int): Number of workers folders that are expected inside the artifacts path.
            n_agents (int): Number of agents in the cluster environment.
        """
        if None in (artifacts_path, n_workers, n_agents):
            pytest.fail("Parameters '--artifacts_path=<path> --n_workers=<n_workers> --n_agents=<n_agents>' are required.")

        # Check if there are threshold data for the specified number of workers and agents.
        selected_conf = f"{n_workers}w_{n_agents}a"
        if selected_conf not in configurations:
            pytest.fail(f"This is not a supported configuration: {selected_conf}. "
                        f"Supported configurations are: {', '.join(configurations.keys())}.")

        # Check if path exists and if expected number of workers matches what is found inside artifacts.
        try:
            cluster_info = ClusterEnvInfo(artifacts_path).get_all_info()
        except FileNotFoundError:
            pytest.fail(f'Path "{artifacts_path}" could not be found or it may not follow the proper structure.')

        if cluster_info.get('worker_nodes', 0) != int(n_workers):
            pytest.fail(f'Information of {n_workers} workers was expected inside the artifacts folder, but '
                        f'{cluster_info.get("worker_nodes", 0)} were found.')

        # Calculate stats from data inside artifacts path.
        data = {'tasks': ClusterCSVTasksParser(artifacts_path).get_stats(),
                'resources': ClusterCSVResourcesParser(artifacts_path).get_stats()}

        if not data['tasks'] or not data['resources']:
            pytest.fail(f'Stats could not be retrieved, "{artifacts_path}" path may not exist, it is empty or it may not'
                        f' follow the proper structure.')

        # Compare each stat with its threshold.
        for data_name, data_stats in data.items():
            for phase, files in data_stats.items():
                for file, columns in files.items():
                    for column, nodes in columns.items():
                        for node_type, stats in nodes.items():
                            for stat, value in stats.items():
>                               th_value = configurations[selected_conf][data_name][phase][file][column][node_type][stat]
E                               KeyError: 'integrity_sync'

tests/performance/test_cluster/test_cluster_performance/test_cluster_performance.py:88: KeyError
============================================================================================ short test summary info =============================================================================================
FAILED tests/performance/test_cluster/test_cluster_performance/test_cluster_performance.py::test_cluster_performance - KeyError: 'integrity_sync'
=============================================================================================== 1 failed in 0.67s ================================================================================================
```

It is related to the `test_cluster_performance` test, the artifacts used to perform this test are the following: [artifacts.zip](https://github.com/wazuh/wazuh-qa/files/8532208/artifacts.zip)

It seems that the error may be related to the way we have to determine the end of the `setup phase` and the beginning of the `stable phase`. In this case, we have that the `setup phase` ends at `12:32:07` according to the `master` log, however, in the `worker8` log we have that `integrity_sync` ends at `12:32:08`.

Therefore, it is possible that this race condition is causing this failure, since we obtain the dates by looking at the master node log: https://github.com/wazuh/wazuh-qa/blob/a86dc639eacec48b74d8bb98eba40cb48207ee0a/deps/wazuh_testing/wazuh_testing/tools/performance/csv_parser.py#L286-L304

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -11,6 +11,10 @@ Release report: TBD
 
 - Update cluster logs in reliability tests ([#2772](https://github.com/wazuh/wazuh-qa/pull/2772))
 
+### Fixed
+
+- Fix an error in the cluster performance tests related to CSV parser ([#2999](https://github.com/wazuh/wazuh-qa/pull/2999))
+
 ## [4.4.0] - Development (unreleased)
 
 Wazuh commit: TBD \

--- a/deps/wazuh_testing/wazuh_testing/tools/performance/csv_parser.py
+++ b/deps/wazuh_testing/wazuh_testing/tools/performance/csv_parser.py
@@ -127,7 +127,7 @@ class ClusterCSVParser:
             for phase in [self.SETUP_PHASE, self.STABLE_PHASE]:
                 for file_name, file_df in files.items():
                     trimmed_df = self._trim_dataframe(file_df, phase, setup_datetime)
-                    if len(trimmed_df):
+                    if len(trimmed_df) and not (phase == self.STABLE_PHASE and file_name == 'integrity_sync'):
                         result[phase][file_name][node] = self._calculate_stats(trimmed_df)
 
         return result

--- a/tests/performance/test_cluster/test_cluster_performance/data/10w_50000a_thresholds.yaml
+++ b/tests/performance/test_cluster/test_cluster_performance/data/10w_50000a_thresholds.yaml
@@ -98,6 +98,6 @@ resources:
           mean: 197162          # (197 MB)
           reg_cof: 209.6
         workers:
-          max: 201480           # (201 MB)
+          max: 256000           # (250 MB)
           mean: 145000          # (145 MB)
           reg_cof: 260

